internal/cli/iam/users/invite.go

```diff
@@ -0,0 +1,227 @@
+
+type InviteOpts struct {
+	cli.OutputOpts
+	username     string
+	password     string
+	country      string
+	email        string
+	mobile       string
+	firstName    string
+	lastName     string
+	orgRoles     []string
+	projectRoles []string
+	store        store.UserCreator
+}
+
+func (opts *InviteOpts) init() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *InviteOpts) createUserView() (*store.UserView, error) {
+	atlasRoles, err := opts.createAtlasRole()
+	if err != nil {
+		return nil, err
+	}
+
+	userRoles, err := opts.createUserRole()
+	if err != nil {
+		return nil, err
+	}
+	user := &store.UserView{
+		AtlasRoles:   atlasRoles,
+		Country:      opts.country,
+		MobileNumber: opts.mobile,
+		User: opsmngr.User{
+			Username:     opts.username,
+			Password:     opts.password,
+			FirstName:    opts.firstName,
+			LastName:     opts.lastName,
+			EmailAddress: opts.email,
+			Roles:        userRoles,
+		},
+	}
+
+	return user, nil
+}
+
+func (opts *InviteOpts) Run() error {
+	user, err := opts.createUserView()
+	if err != nil {
+		return err
+	}
+
+	r, err := opts.store.CreateUser(user)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+const keyParts = 2
+
+func (opts *InviteOpts) createAtlasRole() ([]atlas.AtlasRole, error) {
+	if config.Service() != config.CloudService {
+		return nil, nil
+	}
+
+	atlasRoles := make([]atlas.AtlasRole, len(opts.orgRoles)+len(opts.projectRoles))
+
+	i := 0
+	for _, role := range opts.orgRoles {
+		value := strings.Split(role, ":")
+		if len(value) != keyParts {
+			return nil, fmt.Errorf("unexpected role format: %s", role)
+		}
+		atlasRoles[i] = newAtlasOrgRole(value)
+		i++
+	}
+
+	for _, role := range opts.projectRoles {
+		value := strings.Split(role, ":")
+		if len(value) != keyParts {
+			return nil, fmt.Errorf("unexpected role format: %s", role)
+		}
+		atlasRoles[i] = newAtlasProjectRole(value)
+		i++
+	}
+
+	return atlasRoles, nil
+}
+
+func (opts *InviteOpts) createUserRole() ([]*opsmngr.UserRole, error) {
+	if config.Service() == config.CloudService {
+		return nil, nil
+	}
+
+	if config.Service() != config.CloudService {
+		roles := make([]*opsmngr.UserRole, len(opts.orgRoles)+len(opts.projectRoles))
+
+		i := 0
+		for _, role := range opts.orgRoles {
+			value := strings.Split(role, ":")
+			if len(value) != keyParts {
+				return nil, fmt.Errorf("unexpected role format: %s", role)
+			}
+			roles[i] = newUserOrgRole(value)
+			i++
+		}
+
+		for _, role := range opts.projectRoles {
+			value := strings.Split(role, ":")
+			if len(value) != keyParts {
+				return nil, fmt.Errorf("unexpected role format: %s", role)
+			}
+			roles[i] = newUserProjectRole(value)
+			i++
+		}
+
+		return roles, nil
+	}
+
+	return nil, nil
+}
+
+func newUserOrgRole(role []string) *opsmngr.UserRole {
```

Comment:
[nit] sorry if this was not clear from my previous comment but the idea of passing role, was to do the splitting logic here, `role` as an array of string has no meaning here, either make it a single string and do the split here or makit it two separate `var`s,  `id` and `role`